File: Common/Events/OrderUpdated.cs
using Common.Models;
using Common.Models.Payment;

namespace Common.Events
{
    public record OrderUpdated
    {
        public Guid OrderId { get; init; }

        public PaymentStatus? PaymentStatus { get; set; }

        public CustomerInfo? CustomerInfo { get; init; }

        public OrderStatus? OrderStatus { get; init; }

        public Address? ShippingAddress { get; init; }

        public PaymentInfo? PaymentInfo { get; init; }

        public IList<OrderItem> Items { get; init; } = new List<OrderItem>();
    }
}

File: rorderapi/src/events/orderupdated.rs
use serde::{Deserialize, Serialize};

use crate::models::{address::Address, customerinfo::CustomerInfo, orderitem::OrderItem, orderstatus::OrderStatus, paymentinfo::PaymentInfo, paymentstatus::PaymentStatus};

#[derive(Serialize, Deserialize, Clone, Debug)]
#[serde(rename_all = "PascalCase")]
pub struct OrderUpdated {
    pub order_id: bson::Uuid,
    pub payment_status: PaymentStatus,
    pub customer_info: CustomerInfo,
    pub order_status: OrderStatus,
    pub shipping_address: Address,
    pub payment_info: PaymentInfo,
    pub items: Vec<OrderItem>,
}